backup/src/main/java/io/camunda/zeebe/backup/api/BackupStore.java

```diff
@@ -19,6 +20,12 @@ public interface BackupStore {
   /** Returns the status of the backup */
   CompletableFuture<BackupStatus> getStatus(BackupIdentifier id);
 
+  /**
+   * Uses the given wildcard to list all backups matching this wildcard. The result may be
+   * incomplete.
```

Comment:
For computing the overall status of a backup the impact would be that the overall status is worse than necessary which I would consider good behavior.

For deleting backups, this is also acceptable I'd say.

For marking backups as failed this is not ideal because we need to ensure that after marking as failed, the status can't change back to completed.

WDYT about adding a parameter that disables swallowing exceptions? In that case, listing would fail entirely and marking as failed would too.

Reply:
I don't think the backup that we try to mark as failure will ever be completed. The worse thing that happens is it is never marked as failure, and stay in_progress for ever. This could be confusing for users. But this can happen even if it throws exception. We are not retrying to mark it as failed. 

So it would be ok for now to leave it like this.

Reply:
> I don't think the backup that we try to mark as failure will ever be completed.

Right, but consider this:
1. S3 is partially unavailable
1. listing shows overall status in-progress (or any status other than completed, really) because at least one completed backup was temporarily not found 
2. markFailed finds no backups and thus marks none as failed. 
3. S3 becomes healthy again
4. Listing again finds all backups and shows overall status completed.

This would be surprising at least, incorrect at worst.

Reply:
I don't think it is incorrect. There is no reason why a completed backup should be marked as failed. We want only inprogress ones to be marked as failed. So it is win-win :D